app/uk/gov/hmrc/helptosavereminder/repo/HtsReminderRepository.scala

```diff
@@ -253,6 +244,27 @@ class HtsReminderMongoRepository @Inject()(mongo: ReactiveMongoComponent)
     }
   }
 
+  override def findByCallBackUrlRef(callBackUrlRef: String): Future[Option[HtsUser]] = {
+
+    val tryResult = Try {
+      proxyCollection
+        .find(Json.obj("callBackUrlRef" -> callBackUrlRef), Option.empty[JsObject])
+        .cursor[HtsUser](ReadPreference.primary)
+        .collect[List](maxDocs = 1, err = Cursor.FailOnError[List[HtsUser]]())
+    }
+
+    tryResult match {
+      case Success(s) => {
+        s.map { x =>
+          x.headOption
+        }
+      }
+      case Failure(f) => {
+        Future.successful(None)
+      }
+    }
+  }
+
   override def indexes: Seq[Index] = Seq(
     Index(Seq("nino" -> IndexType.Ascending), Some("nino"), background = true)
```

Comment:
Surely we will need this collection to be indexed by callBackUrlRef

Reply:
Yeah will do.

Reply:
Done